### PR TITLE
Mh 113  admin state requirements

### PR DESCRIPTION
**Pull request by sharrisoncaktus**
Added an inline admin for the requirements to the state admin, and also a separate admin for the requirements themselves in order to enable editing the links.

### CODE REVIEW

**Comment by dsummersl** [comment]
Can one of the admins verify this patch?

File: myhpom/models/state_requirement.py [hunk 1]
```diff
@@ -57,10 +73,28 @@ def state_requirement_pre_save_receiver(sender, instance, **kwargs):
...
+            if (
+                StateRequirement.objects.filter(qid)
+                .filter(state=None, position=instance.position)
+                .exists()
+            ):
+                raise IntegrityError(
+                    '(state, position) not unique: (None, %s)' % (instance.position,)
+                )
+            if StateRequirement.objects.filter(qid).filter(state=None, text=instance.text).exists():
+                raise IntegrityError('(state, text) not unique: (None, %s)' % (instance.text,))
```
Comment (dsummersl) [inline myhpom/models/state_requirement.py]: The 'position' appears to add some unnecessary complexity. As it is the admin interface is hard to use to change the ordering: one must assign some fictional value a StateRequirement, save, then assign the real values and save.

I suggest we remove the position field altogether and rely on pk order.

Neither position/no-position field makes changing the order easy (one requires juggling, the other deleting and re-adding), but I don't think we anticipate this being a regular activity. This model code and the admin interface would be much simpler to maintain and use w/o the position field.

**Review by dsummersl — CHANGES_REQUESTED** [review]
Suggested refinement to the StateRequirement model.

File: myhpom/models/state_requirement.py (hunk 1 above)
Comment (sharrisoncaktus) [reply myhpom/models/state_requirement.py]: Now that we have admin, I agree with you. It's also possible that the admin user can/will change the order by copying/pasting to swap the text of existing entries, which is easier than the existing interface.

**Review by dsummersl — APPROVED** [review]
looks good!

**Pull request merged**